### PR TITLE
refactor: rename `init env` cmd to `init` and remove login cmd

### PR DESCRIPTION
**Pull request by edvald**
This change folds the login handler into the prepareEnvironment
handler. The getEnvironmentStatus handler can now indicate if
prepareEnvironment will need user input, in which case the framework
may error out and guide the user to run `garden init`. This allows
providers to transparently initialize in cases where they don't need
user input, but retain the flexibility to ask for user input when
needed (e.g. for logins, asking for sudo permissions etc.).

### CODE REVIEW

File: examples/remote-k8s/garden.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ project:
   - name: remote
     providers:
     - name: kubernetes
-      context: my-remote-context
+      context: garden-dev-1
```
Comment (edvald) [inline examples/remote-k8s/garden.yml]: This change wasn't meant to go in, will fix along with any other comments

File: garden-cli/src/types/plugin/plugin.ts [hunk 2]
```diff
@@ -103,10 +99,15 @@ export const pluginActionDescriptions: { [P in PluginActionName]: PluginActionDe
...
     description: dedent`
       Check if the current environment is ready for use by this plugin. Use this action in combination
-      with \`prepareEnvironment\` to avoid unnecessary work on startup.
+      with \`prepareEnvironment\`.
 
-      Called before \`prepareEnvironment\`. If this returns \`{ ready: true }\`, the
+      Called before \`prepareEnvironment\`. If this returns \`ready: true\`, the
       \`prepareEnvironment\` action is not called.
+
+      If this returns \`needsUserInput: true\`, the process may throw an error and guide the user to
```
Comment (edvald) [inline garden-cli/src/types/plugin/plugin.ts]: needsUserInput -> needUserInput

File: garden-cli/src/actions.ts [hunk 3]
```diff
@@ -115,35 +112,59 @@ export class ActionHelper implements TypeGuard {
...
+
+    const needUserInput = Object.entries(statuses)
+      .map(([name, status]) => ({ ...status, name }))
+      .filter(status => status.needUserInput === true)
+
+    if (!allowUserInput && needUserInput.length > 0) {
+      const names = needUserInput.map(s => s.name).join(", ")
+      const msgPrefix = needUserInput.length === 1
+        ? `Plugin ${names} has been updated or hasn't been configured, and requires user input.`
+        : `Plugins ${names} been updated or haven't been configured, and require user input.`
```
Comment (thsig) [inline garden-cli/src/actions.ts]: _been updated_ -> _have been updated_

File: garden-cli/src/types/plugin/plugin.ts [hunk 4]
```diff
@@ -126,6 +133,9 @@ export const pluginActionDescriptions: { [P in PluginActionName]: PluginActionDe
     description: dedent`
       Clean up any runtime components, services etc. that this plugin has deployed in the environment.
 
+      Like \`prepareEnvironment\`, this is called sequentially, so handlers are allowed to request user input
```
Comment (thsig) [inline garden-cli/src/types/plugin/plugin.ts]: _called_ -> _executed_?

**Pull request merged**